### PR TITLE
The Withdraw Flow

### PR DESCRIPTION
**Pull request by AtelyPham**
- correct the assset urls in note account table
- feat: update missing dependencies on deposit container
- chore: update withdraw flow state
- add error message for amount validation
- re-style fixed amount items
- add strict to tsconfig for react-hooks package
- add more states for withdraw confirm container
- re-style withdraw confirm component
- correct the relayer in the withdraw confirm card
- fix: the amount calculation
- style: update new design for the note section
- update the disabled logic inside withdraw confirm card
- add missing state inside remove all note function
- set gas limit manually, remove redundant await and cached new leaves
- allow withdraw and unwrap inside unwrap function
- update currency type in withdraw confirm container
- add transaction queue to withdraw flow
- remove notifications
- improve the note account creation animation
- fix: modify gas limit to fix the out of gas transaction error

## Summary of changes
- Add the explorer URL for assets in the note account table
- Re-style components
- Add error message for amount input
- Cached new fetched leaves to improve the leaves fetching function
- Add unwrap functionality into the withdraw function to allows withdraw and unwrap

### Reference issue to close (if applicable)
- Closes https://github.com/webb-tools/webb-dapp/issues/712

-----
### Code Checklist 

- [x] Tested
- [x] Documented

### Additional info

- Right now, I only can deposit and withdraw on the same chain.
- Currently, we don't support withdraw and unwrap that works with relayers
- We also need to improve the transaction state for the `TransactionQueueCard` component to support multiple transactions at the same time.

### Screen Recording

https://user-images.githubusercontent.com/60747384/204351524-dbbec4a8-10dc-4a66-8769-fc53c1d048ee.mp4

### CODE REVIEW

File: libs/web3-api-provider/src/webb-provider/vanchor-withdraw.ts [hunk 1]
```diff
@@ -420,16 +386,39 @@ export class Web3VAnchorWithdraw extends VAnchorWithdraw<WebbWeb3Provider> {
...
+              ...publicInputs,
+              outputCommitments: [
+                publicInputs.outputCommitments[0],
+                publicInputs.outputCommitments[1],
+              ],
+            },
+            extData,
+            unwrapTokenAddress,
+            {
+              gasLimit: 10000000 * 9,
```
Comment (AhmedKorim) [inline libs/web3-api-provider/src/webb-provider/vanchor-withdraw.ts:403]: @dutterbutter  can we have a task for improving gas limit calculation?

Comment (drewstone) [reply libs/web3-api-provider/src/webb-provider/vanchor-withdraw.ts:403]: Gas limit is fixed on these chains so really no need for calculation. Just need to keep a map and update it from time to time (these things update infrequently)

Comment (drewstone) [reply libs/web3-api-provider/src/webb-provider/vanchor-withdraw.ts:403]: Gas price is what changes and could be calculated

**Comment by dutterbutter** [comment]
@AtelyPham please resolve conflicts

File: libs/web3-api-provider/src/webb-provider.ts [hunk 2]
```diff
@@ -266,6 +265,12 @@ export class WebbWeb3Provider
       );
 
       leaves = [...storedContractInfo.leaves, ...leavesFromChain.newLeaves];
+
+      // Cached the new leaves
+      await storage.set(contract.inner.address.toLowerCase(), {
+        lastQueriedBlock: leavesFromChain.lastQueriedBlock,
+        leaves,
+      });
```
Comment (AhmedKorim) [inline libs/web3-api-provider/src/webb-provider.ts:273]: We need to validate these leaves to be right by building the tree and checking the root to be right, this is something we can iterate in a nother PR

File: apps/bridge-dapp/src/hooks/useTransactionProgressValue.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        setProgress(null);
+        break;
+      }
+
+      default: {
+        throw new Error(
+          'Unknown transaction state in DepositConfirmContainer component'
+        );
+      }
+    }
```
Comment (AhmedKorim) [inline apps/bridge-dapp/src/hooks/useTransactionProgressValue.ts:53]: Should be same as Ideal  or removed as we are matching against an enum

**Review by AhmedKorim — CHANGES_REQUESTED** [review]

**Review by AhmedKorim — APPROVED** [review]

**Pull request merged**